### PR TITLE
Allow callouts to be rescheduled from the callback

### PR DESCRIPTION
**Pull request by j-piecuch**
The major change is the addition of the `callout_reschedule()` function, which is used to reschedule a callout from its callback. It can be used to implement interval timers.

### CODE REVIEW

File: sys/kern/callout.c [hunk 1]
```diff
@@ -62,9 +71,19 @@ static void callout_thread(void *arg) {
...
-    callout_clear_active(elem);
-    /* Wake threads that wait for execution of this callout in callout_drain. */
-    sleepq_broadcast(elem);
+
+    WITH_SPIN_LOCK (&ci.lock) {
+      callout_clear_active(elem);
+      if (elem->c_interval) {
+        /* Periodic callout: reschedule. */
+        elem->c_time += elem->c_interval;
+        _callout_schedule(elem);
```
Comment (wiklam) [inline sys/kern/callout.c:80]: It shouldn't happen, but I have been thinking about a situation, when `elem->c_time + n * elem->c_interval < actual time` and we will schedule it n times knowing that the time already passed. Should we worry about such case?

Comment (j-piecuch) [reply sys/kern/callout.c:80]: I don't think rescheduling to the nearest interval multiple in the future is inherently better than rescheduling to a missed deadline: to me it's a matter of choice.

**Review by wiklam — APPROVED** [review]

**Review by pj1031999 — APPROVED** [review]
LGTM

**Review by wiklam — CHANGES_REQUESTED** [review]
Just checking GH Actions.

**Review by wiklam — APPROVED** [review]
LGTM, sorry for the struggle

**Review by wiklam — APPROVED** [review]
LGTM

**Review by cahirwpz — APPROVED** [review]

**Pull request merged**